lib/core/viewer/components/pdfViewer.component.ts

```diff
@@ -133,13 +132,13 @@ export class PdfViewerComponent implements OnChanges, OnDestroy {
         PDFJS.verbosity = 1;
         PDFJS.disableWorker = false;
 
-        let documentContainer = document.getElementById('viewer-pdf-container');
-        let viewer: any = document.getElementById('viewer-viewerPdf');
+        const viewer: any = document.getElementById('viewer-viewerPdf');
 
-        window.document.addEventListener('scroll', this.onDocumentScroll, true);
+        this.documentContainer = document.getElementById('viewer-pdf-container');
+        this.documentContainer.addEventListener('pagechange', (event) => this.onPageChange(event), true);
```

Comment:
You bind anonymous function here, it will never be properly disposed and will leak the memory and degrade over time the more you reopen the viewer. That's why I had this `this.onDocumentScroll = this.onDocumentScroll.bind(this);` and binding/disposing the named function.